qt/src/function.rs

```diff
@@ -492,130 +492,140 @@ pub fn read_directory(arg: &Variant) -> Variant {
 }
 
 #[derive(RustcEncodable)]
-struct FileDetails {
+pub struct FileDetails {
     state: String, // writable, readable, directory, free, inaccessable
     format: Option<String>, // elf, pe, raw
```

Comment:
@flanfly are we considering converting this into a `Option<FileFormat>` and create the corresponding enum?
```rust
enum FileFormat {
    Pe,
    Elf,
    Panop,
    Raw,
}
```

It would make things less arbitrary 😉